packages/lit-dev-content/site/docs/ssr/authoring.md

```diff
@@ -0,0 +1,90 @@
+
+When authoring components, perform imperative DOM operations from lifecycle methods that are called only on the client, and not on the server. See the [lifecycle](#lifecycle) section below for what specific methods can be used.
+
+Some component modules may also have side effects that utilize browser APIs, for example to detect certain browser features, which would break simply by importing in a non-browser environment. If these side-effects cannot be moved to a client-only lifecycle, you must provide proper alternatives or some default behavior for when running in Node.
+
+For simple cases, adding conditionals or optional chaining to certain DOM accesses may be sufficient to guard against unavailable DOM APIs. For example:
+
+```js
+const hasConstructableStylesheets = typeof globalThis.CSSStyleSheet?.prototype.replaceSync === 'function';
+```
+
+The Lit team is also working on providing an easy way of checking whether the code is running in a Node environment which can be utilized to write conditional code blocks targeting different environments. Follow the issue [[labs/ssr] A way for client code to check if running in server (isSsr)](https://github.com/lit/lit/issues/3158) for updates on this feature.
+
+For more complex uses cases, consider utilizing [conditional exports](https://nodejs.org/api/packages.html#conditional-exports) in Node that specifically match for `"node"` environments so you could have different code depending on whether the module is being imported for use in Node or in the browser. Users would get the appropriate version of the package depending on whether it was imported from Node or the browser. Export conditions are also supported by popular bundling tools like [rollup](https://github.com/rollup/plugins/tree/master/packages/node-resolve#exportconditions) and [webpack](https://webpack.js.org/configuration/resolve/#resolveconditionnames) so users can bring in the appropriate code for your bundle.
+
+Note: Due to the way Lit libraries utilize conditional export for providing code meant for import in Node, we strongly discourage bundling `lit` into your component for packages being published to NPM as your bundle will only include a particular version of `lit` meant for a particular environment.
+
+### Lifecycle
+
+Only a select number of web component, `LitElement`, and Lit directive lifecycle callbacks will be run during server-side rendering that are utilized to generate the initial styling and markup for the component. Additional lifecycle methods will be called client-side during hydration and subsequent changes to the page.
+
+The tables below lists the standard custom element and Lit element lifecycle methods and whether they are called during SSR and during hydration.
```

Comment:
This sentence says, "and during hydration," but the tables don't mention hydration at all.

```suggestion
The tables below lists the standard custom element and Lit lifecycle methods and whether they are called during SSR. All of the methods can be called in the browser.
```

One additional, higher-effort possibility would be to *reverse* the sense of the tables. Instead of "Called on server". you could have "Browser-only" with "Yes" and "No*" values. it would require a little rewriting elsewhere (including the suggestion above) but it would mesh nicely with the section above talking about browser-only callbacks. Don't know if it would help any with the concerns @kevinpschaaf expressed.

Reply:
Just made a suggestion here: https://github.com/lit/lit.dev/pull/880#discussion_r954342904

Reply:
Also might make this a little more precise:
```suggestion
The tables below lists the standard custom element and Lit lifecycle methods and whether they are called during SSR. All of the lifecycle is available on the browser after element registration and hydration.
```